File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

# membuat class yang mewarisi class Node dari rclpy, sehingga dapat melakukan semua fungsionalitas di dalamnya
class myNode(Node):
    
    def __init__ (self):
        super().__init__("py_test")
        self.counter_ = 0
        self.get_logger().info("Halo Dunia")
        self.create_timer(1.0, self.timer_callback) # membuat timer

    def timer_callback(self):
        self.get_logger().info("Hello ke-" + str(self.counter_))
        self.counter_ += 1

def main(args=None):
    rclpy.init(args=args) # inisiasi komunikasi ros
    node = myNode() # membuat node dengan nama "py_test" menggunakan class yang diimport dari rclpy.node
    rclpy.spin(node) # membuat node tetap berjalan/ hidup sampai user memencet ctrl+c
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"
#include <chrono>
#include <functional>

class myNode : public rclcpp::Node {
    public:
        myNode() : Node("cpp_test"), counter_(0){
            RCLCPP_INFO(this->get_logger(), "Hello World");
            timer_ = this->create_wall_timer(
                std::chrono::seconds(1),
                std::bind(&myNode::timerCallback, this));
        }

    private:
    void timerCallback() {
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
        counter_++;
    }
        rclcpp::TimerBase::SharedPtr timer_;
        int counter_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv); // inisiasi komunikasi ros2

    // membuat node manual (tanpa OOP)
    // auto node = std::make_shared<rclcpp::Node>("cpp_test");
    // RCLCPP_INFO(node->get_logger(), "Hello world"); 
    
    // membuat node dengan OOP
    auto node = std::make_shared<myNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}